snapcraft/internal/parts.py

```diff
@@ -84,11 +84,11 @@ class _RemoteParts(_Base):
     def __init__(self):
         super().__init__()
 
-        if os.path.exists(self.parts_yaml):
-            with open(self.parts_yaml) as parts_file:
-                self._parts = yaml.load(parts_file)
-        else:
-            self._parts = {}
+        if not os.path.exists(self.parts_yaml):
+            update()
+
+        with open(self.parts_yaml) as parts_file:
+            self._parts = yaml.load(parts_file)
```

Comment:
How many ifs do you want :-P The previous statement makes it available (update). If that fails an exception will be raised already ;-)